core/store/src/flat/chunk_view.rs

```diff
@@ -0,0 +1,46 @@
+use near_primitives::hash::CryptoHash;
+use near_primitives::state::ValueRef;
+
+use crate::Store;
+
+use super::FlatStorage;
+
+/// Struct for getting value references from the flat storage, corresponding
+/// to some block defined in `blocks_to_head`.
+///
+/// The main interface is the `get_ref` method, which is called in `Trie::get`
+/// and `Trie::get_ref` because they are the same for each shard and they are
+/// requested only once during applying chunk.
+// TODO (#7327): lock flat state when `get_ref` is called or head is being updated. Otherwise, `apply_chunks` and
+// `postprocess_block` parallel execution may corrupt the state.
+#[derive(Clone)]
+pub struct FlatStorageChunkView {
+    /// Used to access flat state stored at the head of flat storage.
+    /// It should store all trie keys and values/value refs for the state on top of
+    /// flat_storage.head, except for delayed receipt keys.
+    #[allow(unused)]
+    store: Store,
```

Comment:
Nit: I am somewhat partial to `_store` over `#[allow(unused)]` in this situation. If this field does get used, its easier to notice that its due for a rename, whereas `#[allow(unused)]` might stay here in perpetuity, possibly resulting in this field being maintained even when it is no longer necessary.